### PR TITLE
Guest test td func 1213

### PR DESCRIPTION
**Pull request by hongyuni**

### CODE REVIEW

File: guest-test/tdx/tdx_test_module.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+module_check() {
+  lsmod | grep "$test_module" || \
+  { die "test module $test_module not found in lsmod"; return 1; }
+  test_print_trc "test module loaded and test should be completed now"
+}
+
+###################### Do Works ######################
+if [[ -f "$test_module".ko ]]; then
+  test_print_trc "Kernel test module: $test_module.ko is ready for test"
+  insmod "$test_module".ko || \
```
Comment (ysun) [inline guest-test/tdx/tdx_test_module.sh:28]: Hi Hongyu, it seems that you haven't created the module. I guess it needs to be compiled before inserting it.

File: guest-test/tdx/tdx_halt_test_module/halt_test.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+
+static int __init test_tdx_hlt_init(void)
+{
+	pr_info("[TD guest test] Start to trigger hlt instr.\n");
+	asm("cli");
+	asm("hlt");
+	return 0;
```
Comment (ysun) [inline guest-test/tdx/tdx_halt_test_module/halt_test.c:11]: Is it worthwhile to create a separate kernel module for this? 
I have an idea about creating a common test module in LKVS.
Let's have a discussion later.

**Review by ysun — CHANGES_REQUESTED** [review]

File: guest-test/guest.test_executor.sh [hunk 3]
```diff
@@ -92,7 +92,13 @@ guest_test_close() {
     test_print_trc "guest test complete, close VM now"
     systemctl reboot --reboot-argument=now
 EOF
+ERR_NUM=$?
+if [ $ERR_NUM -eq 0 ]; then
   test_print_trc "Guest VM closed properly after test"
+  return 0
+else
+  return 1
+fi
```
[the remark this answers is not in the document]
Comment (hongyuni) [reply guest-test/guest.test_executor.sh]: ah, my bad (miss-leading by the EOF line before), revised all such issues in this source code.

File: guest-test/tdx/tdx_test_module.sh (hunk 1 above)
Comment (hongyuni) [reply guest-test/tdx/tdx_test_module.sh:28]: there is a test module (the following: guest-test/tdx/tdx_halt_test_module/halt_test.c) being compiled before above test script execution

File: guest-test/tdx/tdx_halt_test_module/halt_test.c (hunk 2 above)
Comment (hongyuni) [reply guest-test/tdx/tdx_halt_test_module/halt_test.c:11]: that would be nice, let's discuss your common test module solution

**Pull request merged**